### PR TITLE
Add large sloped turns to hybrid and single rail

### PR DESCRIPTION
**Pull request by X123M3-256**
This PR adds new track pieces for large sloped turns and diagonal banks to the hybrid and single rail coasters. A corresponding PR has been submitted to the object repo to add the diagonal bank sprites https://github.com/OpenRCT2/objects/pull/236. That one would ideally be merged at the same time or shortly after this one, because those trains will glitch otherwise.

There is a discontinuity caused by the fact that diagonal slopes have significantly shallower gradient than standard slopes. I tried to smooth this out by having the gradient of the turn piece non constant, but it is not possible to remove the discontinuity without making the track piece look distorted. It isn't noticeable with most normal usage but when it is, it's really annoying.

Various diagonal bank transition pieces are also added, these are needed to transition in and out of large banked sloped turns on a diagonal. Previously you could not have sloped banked track on a diagonal.

### CODE REVIEW

**Comment by karst** [comment]
Construction menu seems to no longer grey the light curves when building steep image

The pieces look quite incredible! This is very fun to play with. Thanks for all the effort, you're putting into this :)
I'll make sure to play around a bit more and see if I find any other issues.

**Comment by X123M3-256** [comment]
> Construction menu seems to no longer grey the light curves when building steep

This should be fixed now

**Comment by SpartanFrederic104** [comment]
Please keep this open along with objects#236.

**Comment by duncanspumpkin** [comment]
Seems sensible enough. We will need to increment the park, network and plugin versions for this and add a changelog entry. Could a plugin dev have a quick check that this integrates fine I'm sure it does but best check before hand.

**Comment by duncanspumpkin** [comment]
Plugin Devs have confirmed it works fine

**Comment by X123M3-256** [comment]
Do I need to increase the network version or is this something you do?

**Review by ZehMatt — APPROVED** [review]
Code looks ok, I haven't tested this but the code is mostly additions and not actual changes. There is one comment that is off, otherwise no objections. Also having looked at all changed I noticed that we have plenty of duplicate code and unnamed constants and masks, I will create a few issues for that, not to be addressed in this PR.

**Comment by ZehMatt** [comment]
> Do I need to increase the network version or is this something you do?

Please do.

**Comment by karst** [comment]
Don't forget about the changelog entry :)

**Comment by X123M3-256** [comment]
I have no idea why the Mac build is now failing all I did is update the changelog

**Comment by ZehMatt** [comment]
Hopefully a fluke, I've restarted it, lets see if it completes.

**Comment by karst** [comment]
Seems it was

File: src/openrct2/park/ParkFile.h [hunk 1]
```diff
@@ -9,10 +9,10 @@ struct ObjectRepositoryItem;
 namespace OpenRCT2
 {
     // Current version that is saved.
-    constexpr uint32_t PARK_FILE_CURRENT_VERSION = 22;
+    constexpr uint32_t PARK_FILE_CURRENT_VERSION = 23;
```
Comment (duncanspumpkin) [inline src/openrct2/park/ParkFile.h:12]: unfortunately we just had to bump it to 23 could you rebase one more time and make it 24.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**